airflow/dags/monitor/monitor.py

```diff
@@ -22,6 +28,47 @@
 
 google_service_account_json_value = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS_VALUE", None)
 
+ASK_ASTRO_API_BASE_URL = os.environ.get("ASK_ASTRO_API_BASE_URL", None)
+
+# request ids to monitor from Ask-Astro database.
+ASK_ASTRO_REQUEST_ID_1 = os.environ.get("ASK_ASTRO_REQUEST_ID_1", "05d8882e-56ac-11ee-a818-4200a9fe0102")
+ASK_ASTRO_REQUEST_ID_2 = os.environ.get("ASK_ASTRO_REQUEST_ID_2", "f2d6524c-56ab-11ee-a818-4200a9fe0102")
+
+
+class APIMonitoring:
+    """
+    A class to test API endpoints using GET or POST requests and ensure that they return HTTP 200 status codes.
+    """
+
+    def __init__(self, base_url: str) -> None:
+        self.base_url = base_url
+
+    def test_endpoint(
+        self, endpoint: str, method: str = "GET", data: dict | None = None, headers: dict | None = None
+    ) -> int:
+        """
+        Test an endpoint with the specified method, data, and headers.
+
+        :param endpoint: The endpoint to test.
+        :param method: The HTTP method to use. Defaults to 'GET'.
+        :param data: The data to send in the request. Defaults to None.
+        :param headers: The headers to send in the request. Defaults to None.
+        """
+        try:
+            url = f"{self.base_url}{endpoint}"
+            if method.upper() == "GET":
+                response = requests.get(url, headers=headers)
+            elif method.upper() == "POST":
+                response = requests.post(url, json=data, headers=headers)
+            else:
+                logger.error(f"Unsupported method: {method}")
+                raise Exception(f"Unsupported method: {method}")
+            response.raise_for_status()
+            logger.info(f"Success: {method} {endpoint} returned status code {response.status_code}")
+
+        except requests.exceptions.HTTPError as err:
```

Comment:
why we are not throwing error here?

Reply:
We want to catch that and throw it in slack. I have changed the flow now.

Reply:
https://github.com/astronomer/ask-astro/pull/98/commits/c62341473efd7ad4ffb18de8255d23e73bd154eb